### ISSUE TITLE
sitewide 508-defect-2 [COGNITION, KEYBOARD]: Focus MUST not be hidden by the sticky profile menu

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22824]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a keyboard user using magnification or on a smaller viewport, I always want to be able to see and track focus so I can navigate and interact with the page.

## Details

At smaller viewports or when magnified, the in this section menu will become sticky on the page. This results in it occasionally covering focusable elements. [Keyboard users who rely on visible focus to interact with the page will struggle to do so as shown in research done by the gov.uk team.](https://technology.blog.gov.uk/2018/05/21/sticky-elements-functionality-and-accessibility-testing/)

> Keyboard users often navigate web pages using the TAB key to move through focusable elements, such as links and input boxes. The SHIFT + TAB key combination can be used to move through those elements backwards. We found that where the sticky header overlapped the page, navigating backwards through elements often left the focused element obscured by the sticky element.

> This was an accessibility barrier for users. The sticky element was preventing a commonly used navigation technique from working properly, leaving users unable to see or read the element they had focused on.

> Does this mean that sticky functionality should never be implemented? No, but it must be considered and tested carefully, particularly if the sticky element in any way overlaps other content. A safer use would be a sticky element positioned over an empty part of the page, or to have the content scroll in a way that it is never overlapped by the sticky element.

This is a sitewide issue [that has been resolved before in ticket 19779](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19779) by [referencing the in this section menu on Pittsburgh which is not sticky](https://www.va.gov/pittsburgh-health-care/work-with-us/jobs-careers/).

## Acceptance Criteria

- [ ] Focus is not hidden on the page

## Environment

* Operating System: Any
* Browser: Any (reduce viewport to activate sticky element)
* Screenreading device: Any
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/careers-employment/vocational-rehabilitation/` in browser
2. Reduce screen width to ~600px
3. Tab all the way through the page
4. Shift tab all the way back up the page
5. Confirm focus is hidden by the sticky profile menu

## Proposed Solution 

[This has been resolved before in ticket 19779](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19779) by [referencing the in this section menu on Pittsburgh which is not sticky](https://www.va.gov/pittsburgh-health-care/work-with-us/jobs-careers/).

## WCAG or Vendor Guidance (optional)

* [Gov.uk - Sticky elements functionality and accessibility testing](https://technology.blog.gov.uk/2018/05/21/sticky-elements-functionality-and-accessibility-testing/)
* [WCAG 2.4.7: Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)

## Screenshots or Trace Logs
Focus gets hidden when tabbing back up the page

https://user-images.githubusercontent.com/14154792/113886654-da62f280-978e-11eb-8226-60eadc7427a9.mov

**Comment by tblackwe** [comment]
@Midge-dev Please take a look when you get a chance and determine if we need to resolve.

**Comment by Midge-dev** [comment]
@tblackwe per our slack conversation, this one can go to va-platform-cop-frontend

**Comment by Midge-dev** [comment]
@tblackwe I suggested this be reviewd by va-platform-cop-frontend team because of the "In this section component" seen in the video above. I think they're the ones who will decide whether or not this component becomes "non-sticky" sitewide

**Comment by tblackwe** [comment]
@steele-lm When we get understanding on how to reassign/link tickets to forms/design teams, this ticket will need to follow that process. blocking it until then.

**Comment by steele-lm** [comment]
Closing this. Handed off to Platform team and subsequently closed.
